### DISCUSSION TITLE
Function & expression in ParsedMaterial

### DISCUSSION THREAD

**Discussion by hokonkim** [idaholab/moose#19040] [Q&A General] [answered]
Hello, I'm trying to use both function and expression(if-else) in "function in ParsedMaterial" like below

However, it doesn't work..

(*** ERROR ***
Invalid function
if(aux_eta_tpb > funcOverpotential, 0, 5318)
aux_eta_tpb
in ParsedMaterialHelper.
)

How can I use "funcOverpotential function" in ParsedMaterial? 

Thanks.

    [Functions]
      [./funcPotentialLSM]
        type = ParsedFunction
        value = 'init_val := if(t = 0, 0.03607, 0);
                 init_val +BC_YSZ +E_rev - eta * t'
        vars  = '          BC_YSZ  E_rev   eta'
        vals  = 'numerical_phi_YSZ 1.03665 0.4' # 0.03607+1.03665
      [../]

      [./funcOverpotential]
        type = ParsedFunction
        value = 'eta * t'
        vars  = 'eta'
        vals  = '0.4'
      [../]

      [./funcTimeStepper]
        type = PiecewiseLinear
        x = '0.0  0.1 0.5 1.0' # time_t
        y = '0.01 0.1 0.1 0.2' # time_dt
      [../]
    []

    [Materials]
      [./tpbActivity]
        type  = ParsedMaterial
        block = 'PT_TPB_TET4'
        f_name = 'tpbActivity_S0'
        function = 'if(aux_eta_tpb > funcOverpotential, 0, 5318)'
        args = 'aux_eta_tpb'
        # Make values see at output(*.e) file
        outputs = exodus
      [../]
    []

**Comment by GiudGiud** [comment, marked as answer]
Hello

ParsedMaterial cannot use a function directly (funcOverpotential). 
You have to use a GenericFunctionMaterial to transfer the function to a material property. Then make sure to have it listed in `material_property_names`

Guillaume
